core/src/main/java/org/apache/sling/testing/mock/osgi/context/OsgiContextImpl.java

```diff
@@ -133,75 +133,73 @@ private void registerDefaultServices() {
     }
 
     /**
-     * Injects dependencies, activates and registers a service in the mocked OSGi environment.
-     * @param <T> Service type
-     * @param service Service instance
-     * @return Registered service instance
+     * Injects dependencies, activates and registers a DS component in the mocked OSGi environment.
+     * Construction injection for OSGi services is supported.
+     * @param <T> DS Component type
+     * @param component a DS component instance
+     * @return the DS component instance
      */
-    public final @NotNull <T> T registerInjectActivateService(@NotNull final T service) {
-        return registerInjectActivateService(service, (Map<String,Object>)null);
+    public final @NotNull <T> T registerInjectActivateService(@NotNull final T component) {
```

Comment:
The name is a bit weird as this may potentially register multiple services but I guess it is too late now to change that. Maybe we can deprecate these method names and instead introduce `injectActivateRegisterComponent` (because in fact service registration happens after calling activate). WDYT @stefanseifert

Reply:
or alternatively just `setupComponent(...)` or `setupDsComponent(...)`

Reply:
i would prefer updating the comments if required than renaming the method - the name is a bit awkward but used in nearly every unit test over all those years, it think there is no benefit renaming it now (although i like the simplicity of "setupDsComponent")